runtime/common/FmtCore.h

```diff
@@ -0,0 +1,16 @@
+/*************************************************************** -*- C++ -*- ***
+ * Copyright (c) 2022 - 2023 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ *******************************************************************************/
+
+#if (defined(__GNUC__) && !defined(__clang__) && !defined(__INTEL_COMPILER))
```

Comment:
One thing to note maybe - These warnings are specific to GCC 12.2. New version (major or minor) will not require this.

Reply:
Interesting. I just checked with the gcc-12 version that the ubuntu 22.04 package manager installs. They did seem unnecessary to me.